### ISSUE TITLE
Unexpected behavior with LeveledCompactionStrategy

### ISSUE THREAD

**Issue by glommer** [scylladb/scylla#3279]
Scylla HEAD: 78a9ab827e05563358d98e21fe1c523103393b4c
Seastar HEAD: bcfbe0c3f7f62af97d58bef4383d46864b32fbe7

I have ran a simple cassandra-stress on a previously empty single-node cluster smp=1 for 10 minutes:

```
cassandra-stress write duration=10m cl=ONE no-warmup \
    -pop seq=1..10000000000000 \
    -schema compaction\(strategy='LeveledCompactionStrategy'\) \
    -mode native cql3 -rate threads=30 limit=50000/s -node 127.0.0.1 \
```

my expectation is that, because there are no manual compactions, no SSTables coming in our out, just LeveledCompaction doing its thing, every compaction would already generate SSTables respecting the Level invariants.

That's not what I see in the logs:
```
INFO  2018-03-13 09:47:20,298 [shard 0] LeveledManifest - Leveled compaction strategy is restoring invariant of level 1 by compacting 16 sstables on behalf of keyspace1.standard1
```

I see plenty of messages like those. I can reproduce this every time.

**Comment by raphaelsc** [comment]
Definitely a bug. I'll take a look. Thanks!

**Comment by raphaelsc** [comment]
I'd bet it was probably introduced after changes in compaction manager for postponed compaction reevaluation.

**Comment by glommer** [comment]
guess that makes it a regression ?

On Tue, Mar 13, 2018 at 11:37 AM, Raphael Carvalho <notifications@github.com

**Comment by raphaelsc** [comment]
yes. if invariant is broken, it very probably means two compaction was running in parallel. care to share the log?

**Comment by glommer** [comment]
[log-leveled.txt](https://github.com/scylladb/scylla/files/1807611/log-leveled.txt)

**Comment by raphaelsc** [comment]
my theory holds. i'll work on fixing it today. thanks for the report.

**Comment by glommer** [comment]
:+1: please send backports to the versions affected as well (if needed)

**Comment by glommer** [comment]
@slivne 2.2 blocker.
